src/main/java/org/jitsi/videobridge/xmpp/MediaStreamTrackFactory.java

```diff
@@ -383,157 +106,159 @@ private static void removeReferences(TrackSsrcs trackSsrcs, List<SourcePacketExt
+            {
+		SourceGroupPacketExtension sourceGroup = it.next();
+		if (sourceGroup.getSemantics().equalsIgnoreCase(
+			    SourceGroupPacketExtension.SEMANTICS_SIMULCAST))
+		{
+                    addVP8SimulcastTrack(sourceGroup);
+                    
+                    // We don't want to process this group EVER again!
+                    it.remove(); 
+                }
+	    }
 
-    /**
-     * Describes a single secondary ssrc and its group semantics to
-     * its primary ssrc (e.g. FID or FEC-FR)
-     */
-    private static class SecondarySsrc
-    {
-        public long ssrc;
-        public String type;
+            // Now we're only left with RTX and FEC groups.
+            for (SourceGroupPacketExtension sourceGroup : groupsCopy)
+            {
+                List<SourcePacketExtension> groupSources
+                    = sourceGroup.getSources();
 
-        public SecondarySsrc(long ssrc, String type)
-        {
-            this.ssrc = ssrc;
-            this.type = type;
-        }
-    }
+                if (groupSources.size() != 2)
+                {
+                    logger.warn(
+                            "A valid RTX or FEC group has exactly 2 ssrcs.");
+                    continue;
+                }
 
-    /**
-     * Groups a set of secondary ssrcs for a single primary ssrc (e.g. its
-     * rtx and fec ssrcs)
-     */
-    private static class SecondarySsrcs
-        implements Iterable<SecondarySsrc>
-    {
-        public List<SecondarySsrc> secondarySsrcs;
+                long primarySSRC = groupSources.get(0).getSSRC();
+    
+                MediaStreamTrackDesc track = addPlainTrack(primarySSRC);
 
-        public SecondarySsrcs(List<SecondarySsrc> secondarySsrcs)
+                // Set the secondary SSRC for the matching encodings.
+                long secondarySSRC = groupSources.get(1).getSSRC();
+                tracksMap.put(secondarySSRC, track);
+                for (RTPEncodingDesc encoding : track.getRTPEncodings())
+                {
+                    if (encoding.getPrimarySSRC() == primarySSRC)
+                    {
+                        encoding.addSecondarySsrc(
+                                secondarySSRC,
+                                semanticsMap.get(sourceGroup.getSemantics()));
+                    }
+                }
+            }
+	}
+	
+        // Process the free ssrcs
+        if (sources != null && !sources.isEmpty())
         {
-            this.secondarySsrcs = secondarySsrcs;
+            for (SourcePacketExtension source : sources)
+            {
+                long primarySSRC = source.getSSRC();
+                addPlainTrack(primarySSRC);
+            }
         }
 
-        @Override
-        public Iterator<SecondarySsrc> iterator()
-        {
-            return secondarySsrcs.iterator();
-        }
+        Collection<MediaStreamTrackDesc> tracks = tracksMap.values();
+	return tracks.toArray(new MediaStreamTrackDesc[tracks.size()]);
```

Comment:
Above we map the primary SSRC and the secondary SSRC to their track. So tracksMap's values might contain duplicates. Is this intentional?